src/ui/windows/TogglDesktop/TogglDesktop/ui/windows/MainWindow.xaml.cs

```diff
@@ -391,6 +392,24 @@ private void onLogin(bool open, ulong userID)
             this.updateTracking(null);
         }
 
+        private void OpenSSOLoginView()
+        {
+            setActiveView(SSOLoginView);
+        }
+
+        private void OpenLoginView(string confirmationCode, string email)
+        {
+            if (!string.IsNullOrEmpty(confirmationCode))

```

Comment:
I think this logic should in principle be handled by the LoginView or LoginViewModel rather than the MainWindow